### PR TITLE
My code review suggestions.

### PR DESCRIPTION
**Pull request by bdaha**
Changes :
Stock DTO : added nullability in the constructors to be cautious of any potential bugs.
MainFragment.kt : Put the lambda expression in one like to save space.
RetrofitClientInstance: Added a const in the uppercase underscore-separated name.

Analysis of the program - This program is for an app that assists in stock trading. The program has the main fragment UI that compiles and runs. The autocomplete also works as it should. naming conventions are used correctly.
Was the program available in UC Github on time? - Yes
Is the program documented/commented well enough for you to understand? - Yes
Does the program compile? - Yes
Rationale behind your changes - I applied what we learned in the class videos along with some articles.

Technical concepts that I learned:
1. When I was researching the changes I could make, I learned about the modifier orders. I learned that if a declaration has multiple modifiers, they should be in a specific order.

 Sample - private const val BASE_URL = "https://gist.githubusercontent.com/" - This line of code shows the modifiers in correct Order.

2. While researching, I also learned more about the init constructors. When I used the init constructor before, I was just following along to the code presented and didn't really think about what it was. With this article, https://blog.mindorks.com/understanding-init-block-in-kotlin I was able to understand what 

sample - class MainViewModel : ViewModel() {
    var stocks: MutableLiveData<ArrayList<Stock>> = MutableLiveData<ArrayList<Stock>>()
    var stockService: StockService = StockService()

    /**
     * Initialize the MVM.
     */
    init {
        fetchStocks()
    }
On here the init function is under the primary constructor and it is initialized properly.

3. Another new concept I learned about was @SerializedName

sample - On this block of code, the group used @serializedname and used the "value:" in a way that was new for me.

    @SerializedName("symbol") var stockTicker: String?,
    @SerializedName("company") var stockCompany: String?,
    @SerializedName("description") var stockDescription: String?,
    @SerializedName("initial_price") var stockPrice: String?

### CODE REVIEW

File: app/src/main/java/edu/uc/coffeens/stocktracker/dto/Stock.kt [hunk 1]
```diff
@@ -6,10 +6,10 @@ import com.google.gson.annotations.SerializedName
...
  */
 data class Stock(
-    @SerializedName("symbol") var stockTicker: String,
-    @SerializedName("company") var stockCompany: String,
-    @SerializedName("description") var stockDescription: String,
-    @SerializedName("initial_price") var stockPrice: String
+    @SerializedName("symbol") var stockTicker: String?,
+    @SerializedName("company") var stockCompany: String?,
+    @SerializedName("description") var stockDescription: String?,
+    @SerializedName("initial_price") var stockPrice: String?
```
Comment (bdaha) [inline app/src/main/java/edu/uc/coffeens/stocktracker/dto/Stock.kt:12]: I added the question marks to make sure the code can be nulled in the future to be ready against any potential bugs that might occur, as advised by this article https://docs.sonarqube.org/display/SONARQUBE45/Bugs+and+Potential+Bugs .

File: app/src/main/java/edu/uc/coffeens/stocktracker/services/RetrofitClientInstance.kt [hunk 2]
```diff
@@ -9,7 +9,7 @@ import retrofit2.converter.gson.GsonConverterFactory
 object RetrofitClientInstance {
 
     private var retrofit: Retrofit? = null;
-    private val BASE_URL = "https://gist.githubusercontent.com/"
+    private const val BASE_URL = "https://gist.githubusercontent.com/"
```
Comment (bdaha) [inline app/src/main/java/edu/uc/coffeens/stocktracker/services/RetrofitClientInstance.kt:12]: In an attempt to make sure the code's modifiers were descriptive as possible, I added the const modifier as specified in here https://kotlinlang.org/docs/coding-conventions.html#property-names

File: app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -36,13 +36,7 @@ class MainFragment : Fragment() {
...
         viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
         viewModel.stocks.observe(viewLifecycleOwner, Observer { stocks ->
-            actStock.setAdapter(
-                ArrayAdapter(
-                    context!!,
-                    R.layout.support_simple_spinner_dropdown_item,
-                    stocks
-                )
-            )
+            actStock.setAdapter(ArrayAdapter(context!!, R.layout.support_simple_spinner_dropdown_item, stocks))
```
Comment (bdaha) [inline app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainFragment.kt:39]: To make sure the code looks "less" as possible I put the lambda expression in one line. Even though the code before used good trailing commas, lambdas can help make the space around code be smaller, as discussed in the lecture videos and this article https://kotlinlang.org/docs/coding-conventions.html#lambdas

File: app/src/main/java/edu/uc/coffeens/stocktracker/dto/Stock.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/coffeens/stocktracker/dto/Stock.kt:12]: Actually... best to keep them not nullable.  ? opens up the opportunity that they can be null; no ? means they can't .  I'd leave this as it were.

File: app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainFragment.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainFragment.kt:39]: Either way works.

**Comment by discospiff** [comment]
I'm neutral on merging this branch... leave it up to the group to decide on whether or not to merge or harvest.

**Pull request closed without merging**